mdpow/workflows/dihedrals.py

```diff
@@ -544,30 +614,65 @@ def plot_violins(df, resname, figdir=None, molname=None, width=0.9, solvents=SOL
     if molname is None:
         molname = resname
 
+        # figdir is now necessary??
     if figdir is not None:
         subdir = molname
         newdir = os.path.join(figdir, subdir)
         os.mkdir(newdir)
 
-    section = df.groupby(by='selection')
+    section = df.groupby(by="selection")
 
+    # conversion factor: 1 mm = 3.7795275591 px
+    plot_pdf_width = plot_pdf_width * 3.7795275591
+    # DEFAULT: 190 mm = 718.110236229 pixels
+
+    # put this into separate function
     for name in section:
+        a = list(ab_pairs[name[0]][0])
+        b = list(ab_pairs[name[0]][1])
+
+        drawer = rdMolDraw2D.MolDraw2DSVG(250, 250)
+        drawer.DrawMolecule(mol=mol, highlightAtoms=a, highlightBonds=b)
+        drawer.FinishDrawing()
+        svg = drawer.GetDrawingText().replace('svg:','')
+
+        mol_svg = sg.fromstring(svg)
+        m = mol_svg.getroot()
+        m.scale(0.0125).moveto(21.8, 0.35)
+
         plot = dihedral_violins(name[1], width=width, solvents=solvents)
-        plot.set_titles(f'{molname},{name[0]}, ''{col_name}')
+        _ = plot.set_titles(f'{molname}, {name[0]} {list(ab_pairs[name[0]][0])} | ''{col_name}')
```

Comment:
Should work without the dummy assignment.

Reply:
I'd move `set_titles` to inside `dihedral_violins(..., title=f'{molname}, {name[0]} {list(ab_pairs[name[0]][0])} | ''{col_name}')` with a default of `None`. In this way, all plot styling is contained inside the function.

Reply:
And then you can also remove 4 lines of comment.